### PR TITLE
A few additional drive utils

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/drive.py [hunk 1]
```diff
@@ -60,24 +61,32 @@ def getPathNorm(self, path):
...
         return [reqValidName(p.strip().lower()) for p in path]
 
-    def getItemInfo(self, iden):
-        return self._getItemInfo(s_common.uhex(iden))
+    def getItemInfo(self, iden, typename=None):
+        info = self._getItemInfo(s_common.uhex(iden))
+        if typename is not None and info.get('type') != typename:
+            infotype = info.get('type')
+            mesg = f'Drive item has the wrong type. Expected: {typename} got {infotype}.'
+            raise s_exc.TypeMismatch(mesg=mesg)
```
Comment (vEpiphyte) [inline synapse/lib/drive.py:69]: Do we want to include `typename` and `infotype` keys here as well?

File: synapse/lib/drive.py [hunk 2]
```diff
@@ -60,24 +61,32 @@ def getPathNorm(self, path):
...
+    def _reqItemInfo(self, bidn, typename=None):
         info = self._getItemInfo(bidn)
-        if info is not None:
-            return info
+        if info is None:
+            mesg = f'No drive item with ID {s_common.ehex(bidn)}.'
+            raise s_exc.NoSuchIden(mesg=mesg)
+
+        if typename is not None and info.get('type') != typename:
+            raise s_exc.SynErr(mesg='Drive item has the wrong type.')
```
Comment (vEpiphyte) [inline synapse/lib/drive.py:87]: Typemismatch  ?

File: synapse/lib/drive.py (hunk 1 above)
Comment (invisig0th) [reply synapse/lib/drive.py:69]: We never do anything with them, so *shrug*.

File: synapse/lib/drive.py (hunk 2 above)
Comment (invisig0th) [reply synapse/lib/drive.py:87]: thanks, the TypeMismatch exc was a last minute add :+1:

File: synapse/lib/drive.py [hunk 3]
```diff
@@ -494,10 +503,27 @@ def getTypeSchema(self, typename):
...
 
+        if vers is not None:
+            vers = int(vers)
+            curv = self.getTypeSchemaVersion(typename)
+            if curv is not None:
+                if vers == curv:
+                    return False
+
+                if vers < curv:
+                    mesg = 'Cannot downgrade drive schema version.'
```
Comment (vEpiphyte) [inline synapse/lib/drive.py:524]: Should we include the typename of the scheme being "downgraded" ?

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**